Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -18,276 +17,286 @@
-        headers = {
-            'Authorization': 'Bearer ' + token
-        }
-        response = self._http_request(
-            method='GET',
-            headers=headers,
-            params={'q': time_param}
+
+        return response['access_token']
+
+    def export_events(self, export_profile: str) -> dict:
+        """
+        Exports events for the export profile that arrived after setting the internal bookmark.
+
+        Args:
+            export_profile (str): Export profile name.
+
+        Returns:
+            dict: Response JSON.
+        """
+        return self._http_request(
+            method='POST',
+            url_suffix=f'/rest/2.0/export_profiles/{export_profile}/export',
+            raise_on_status=True,
+        )
+
+    def set_export_bookmark(self, export_profile: str) -> str:
+        """
+        Advances the internal bookmark / pointer for the export profile based on the last export events.
+        This is effectively an API-side implementation of `demisto.getLastRun` and `demisto.setLastRun`.
+
+        *Only call after successfully exporting events and sending them to XSIAM!*
+
+        Returns:
+            str: Response text.
+        """
+        return self._http_request(
+            method='POST',
+            url_suffix=f'/rest/2.0/export_profiles/{export_profile}/acknowledge',
+            raise_on_status=True,
+            resp_type='text',
         )
-        return response
 
 
-def test_module(client: Client, params: Dict[str, Any]) -> str:
+def test_module(client: Client, export_profiles: list[str]) -> str:
     """
-    Tests API connectivity and authentication'
-    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
-    successful.
-    Raises exceptions if something goes wrong.
+    Tests API connectivity and authentication and validates export profiles.
     Args:
         client (Client): Digital Guardian client to use.
-        params (Dict): Integration parameters.
+        export_profiles (list): List of export profile names.
     Returns:
-        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+        str: 'ok' connection to the service is successful.
+    Raises:
+        DemistoException | HTTPError: If request failed.
     """
-    try:
-        limit = arg_to_number(params.get('number_of_events')) or 1000
-        fetch_events(
-            client=client,
-            last_run={},
-            limit=limit,
-        )
+    invalid_export_profiles = set()
 
-    except Exception as e:
-        if 'Forbidden' in str(e):
-            raise DemistoException('Authorization Error: make sure API Key is correctly set')
-        else:
-            raise e
-    return "ok"
+    for export_profile in export_profiles:
+        try:
+            client.export_events(export_profile)
 
+        except Exception as e:
+            if 'Forbidden' in str(e):
```

Comment:
Checking for a string is usually not very reliable, can we check the http code instead?

Reply:
Updated logic in commit https://github.com/demisto/content/commit/de6eca13f444da175cf25459c9d3a91d521fdc81

Checking the HTTP status code isn't particularly straight forward since `Client._http_request` raises a custom exception (`DemsitoException`). I have done it be accessing `DemistoException.exception.response.status_code`

Please review the changes in the commit and let me know if you want additional improvments.

Reply:
Updated logic in commits https://github.com/demisto/content/commit/de6eca13f444da175cf25459c9d3a91d521fdc81 and https://github.com/demisto/content/commit/94504176a4a210721c2530b783f5375765a6190b